### DISCUSSION TITLE
Android 5.1 Termux 118 build process

### DISCUSSION THREAD

**Discussion by Xinul-Hacker** [termux/termux-app#3038] [Q&A] [answered]
Can I please get a brief overview on how to build an APK (or package) for Android 5.1?
I'm trying to build gdb with python support so that I can use gef with it.
I've reviewed the documents and will continue to work to understand the build process, but a link to a little clearer guidance would be great.
BTW, I'm building on termux 118 from Android 10 for an offline device running Android 5.1 which does not have internet access capabilities.

**Comment by agnostic-apollo** [comment, marked as answer]
To build apk, pull `termux-app` `master` branch and change `packageVariant` to `apt-android-5`.

https://github.com/termux/termux-app/blob/3f7a93931329d4d5cd10a8ec45eb29126984bd5e/app/build.gradle#L12

Builds are already available from actions.

https://github.com/termux/termux-app/wiki/Termux-on-android-5-or-6

To build packages, use `android-5` `termux-packages` branch.

https://github.com/termux/termux-packages/tree/android-5

https://github.com/termux/termux-packages/wiki/Building-packages
